### PR TITLE
DAOS-13813 ci: Move GHA build to Fedora 38

### PR DESCRIPTION
**Pull request by jolivier23**
Fix a couple of newer compiler warnings on Fedora 38
Move developer build to ucx 1.14.1.   The RPM update can be done
independently or not at all.   We only build the RPM to build DAOS.
We use MOFED drivers when we actually use UCX at runtime.

Required-githooks: true

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

File: utils/build.config [hunk 1]
```diff
@@ -10,7 +10,7 @@ SPDK = v22.01.2
 OFI = v1.18.0
 MERCURY = v2.3.0
 PROTOBUFC = v1.3.3
-UCX=v1.13.0
+UCX=v1.14.1
```
Comment (ashleypittman) [inline utils/build.config:13]: This isn't mentioned in the commit message, does anything else on the rpm side need a corresponding change?

Comment (jolivier23) [reply utils/build.config:13]: It's in the follow-on commit message.  I couldn't get 1.13.0 to compile on my fedora 38 box.  The RPM isn't actually used in any testing but there has been an RPM update in progress for a while.  My main point in doing this was that there are no compiler warnings.   Locally, I've just had WARNING_LEVEL=warn set.

**Review by ryon-jensen — APPROVED** [review]

**Review by ashleypittman — DISMISSED** [review]

**Review by daltonbohning — APPROVED** [review]

File: utils/build.config (hunk 1 above)
Comment (jolivier23) [reply utils/build.config:13]: We don't use this version of UCX in testing.  It's just for the build and then MOFED is used at runtime.   So I'm not sure it matters much.  That said, @jgmoore-or does have an older update PR in progress: https://github.com/daos-stack/openucx/pull/9

**Review by jiafuzha — DISMISSED** [review]
LGTM

File: site_scons/prereq_tools/base.py [hunk 2]
```diff
@@ -721,6 +721,7 @@ def define(self, name, **kw):
             extra_include_path -- Subdirectories to add to dependent component path
             out_of_src_build -- Build from a different directory if set to True
             build_env -- Environment variables to set for build
+            skip_arch -- not required on this architecture
```
Comment (ashleypittman) [inline site_scons/prereq_tools/base.py:724]: I'm not convinced by the name of this flag, there are many reasons why components might define something to be empty.

**Review by ashleypittman — APPROVED** [review]

**Review by daltonbohning — APPROVED** [review]

**Pull request merged**